### ISSUE TITLE
Save metadata field by pressing Enter

### ISSUE THREAD

**Issue by melissachang** [adobe-dmeservices/custom-metadata#22]
- Edit metadata field, such as Description
- [Desired] To save, I would like to simply press Enter. This is how Lightroom Classic works.
- [Actual] I have to click `Save Changes` button

This extra mouse click gets in the way, when I'm editing a bunch of photos.

**Comment by jameslockman** [comment]
Thanks for this detail, Melissa. We explored variations on this theme when designing the interface. Bridge has a model where it will automatically apply changes after a period of inactivity on a field, and it also has a pencil icon where a user can commit change more quickly. We wanted users to be able to make bulk changes and then commit them at once, which allows us to include a revert/undo procedure that many users find invaluable. 
In addition, we support multi-line entry, so using the Enter or Carriage Return would interfere with multi-line data entry. 
We can consider a modified Enter (Control/Command-Enter) as an alternative, however.

**Comment by melissachang** [comment]
Thanks for the details. Sounds like I can skip clicking `Save Changes` button, and description will eventually automatically be saved.

`Enter to save` is much more common that `Enter to add newline`, so that should be the default. That's what many UIs do. For example:

- In Slack, Enter sends messaage. Shift-Enter adds newline.
- OTOH in GitHub issues, `Enter to add newline` is more common, so that is the default.

**Comment by jameslockman** [comment]
No, Bridge offers this, but our panel requires that you click on the Save Changes button. Your changes will not be saved in the Custom Metadata Panel unless you click Save Changes.

**Comment by willwenzel** [comment]
Hi @jameslockman, I'd like to upvote this request, and your idea of a modified enter would be a huge help to those of us who type a lot of metadata!

**Comment by willwenzel** [comment]
Actually @jameslockman,  if there were a perfect solution, with only keyboard shortcuts one could save changes, select the next image in bridge, and then select the first editable field in Custom Metadata.  Just in case you wanted to make it ideal for data entry!

**Comment by jameslockman** [comment]
Thanks for this feedback. This UI behavior remains at the top of the backlog.

In the meantime, we discovered that there is a bug in Bridge that prevents us from registering interest in specific keystrokes, such as enter. Once that gets fixed (maybe in the upcoming release around MAX timeframe), we can revisit.

**Comment by jameslockman** [comment]
Bridge team let us know that they plan to pick up the key registration issues after MAX 2024. Fingers crossed!
